lightwood/encoders/time_series/rnn.py

```diff
@@ -149,26 +197,76 @@ def encode(self, column_data, get_next_count = None):
         else:
             return self._pytorch_wrapper(torch.stack(ret)), self._pytorch_wrapper(torch.stack(next))
 
+    def _decode_one(self, hidden):
+        """
+        Decodes a single time series from its encoded representation.
+        :param hidden: time series embedded representation tensor, with size self._encoded_vector_size
+        :return: decoded time series list
+        """
+        self._decoder.eval()
+        with torch.no_grad():
+            ret = []
+            next_tensor = torch.full((1, 1, self._n_dims), self._sos, dtype=torch.float32).to(self.device)
+            for _ in range(self._max_dec_length):
+                next_tensor, hidden = self._decoder.forward(next_tensor, hidden)
+                ret.append(next_tensor)
+            return torch.stack(ret)
+
+    def decode(self, encoded_data):
+        """
+        Decode a list of embedded multidimensional time series
+        :param encoded_data: a list of embeddings [ e1, e2, ...] to be decoded into time series
+        :return: a list of reconstructed time series
+        """
+        if not self._prepared:
+            raise Exception('You need to call "prepare_encoder" before calling "encode" or "decode".')
 
+        ret = []
+        for _, val in enumerate(encoded_data):
+            hidden = torch.unsqueeze(torch.unsqueeze(val, dim=0), dim=0).to(self.device)
+            reconstruction = self._decode_one(hidden).cpu().squeeze().T.tolist()
+            ret += [reconstruction]
 
-# only run the test if this file is called from debugger
-if __name__ == "__main__":
-    series = []
-    for j in range(100):
-        length = 3 + np.random.randint(14)
-        skip = 1+np.random.randint(4)
-
-        start = np.random.randint(30)
-        vec = [start+j*skip for j in range(length)]
-
-        series+=[' '.join([str(x) for x in vec])]
-
-    encoder = RnnEncoder(encoded_vector_size=3,train_iters=10)
-    encoder.prepare_encoder(series, feedback_hoop_function=lambda x:print(x))
-
+        return self._pytorch_wrapper(ret)
 
-    # test de decoder
-    init_vector = ['31 33 35 37', '1 2 3 4 5 6']
 
-    print(encoder.encode(column_data=init_vector, get_next_count=2))
-    print(encoder.encode(column_data=init_vector))
+# only run the test if this file is called from debugger
+if __name__ == "__main__":
```

Comment:
The tests should be inside the `tests`  dir, specifically here: https://github.com/mindsdb/lightwood/tree/6c1d250e04d2978e09af854175d8fd8e4c78156a/tests/unit_tests

under a new dir `encoders/time_series`

... Sorry, new change, still not a thing for the other encoders, but we're trying to separate the tests from code atm, should be done for the old ones, but for now we can do is to start doing it for new code.